src/transformers/exports.ts

```diff
@@ -0,0 +1,206 @@
+      });
+    }
+
+    // TODO(KB): Sourcemaps fail :(
+    return {
+      code,
+    };
+  }
+
+  /**
+   * After Closure Compiler has modified the source, we need to replace the window scoped
+   * references we added with the intended export statements
+   * @param code source post Closure Compiler Compilation
+   * @param id Rollup identifier for the source
+   * @return Promise containing the repaired source
+   */
+  public async postCompilation(code: string, id: string): Promise<TransformSourceDescription> {
+    if (this.outputOptions === null) {
+      this.context.warn(
+        'Rollup Plugin Closure Compiler, OutputOptions not known before Closure Compiler invocation.',
+      );
+    } else if (this.outputOptions.format === 'es') {
+      const exportedConstants: Array<string> = [];
+
+      Object.keys(this.exported).forEach(key => {
+        switch (this.exported[key]) {
+          case ExportClosureMapping.NAMED_FUNCTION:
+            code = code.replace(`window.${key}=function`, `export function ${key}`);
+            break;
+          case ExportClosureMapping.NAMED_CLASS:
+            const namedClassMatch = new RegExp(`window.${key}=(\\w+);`).exec(code);
+            if (namedClassMatch && namedClassMatch.length > 0) {
+              // Remove the declaration on window scope, i.e. `window.Exported=a;`
+              // Replace it with an export statement `export {a as Exported};`
+              code = code.replace(namedClassMatch[0], `export {${namedClassMatch[1]} as ${key}};`);
+            }
+            break;
+          case ExportClosureMapping.NAMED_DEFAULT_FUNCTION:
+            code = code.replace(`window.${key}=function`, `export default function ${key}`);
+            break;
+          case ExportClosureMapping.NAMED_DEFAULT_CLASS:
+            const namedDefaultClassMatch = new RegExp(`window.${key}=(\\w+);`).exec(code);
+            if (namedDefaultClassMatch && namedDefaultClassMatch.length > 0) {
+              // Remove the declaration on window scope, i.e. `window.Exported=a;`
+              // Replace it with an export statement `export default a;`
+              code = code.replace(
+                namedDefaultClassMatch[0],
+                `export default ${namedDefaultClassMatch[1]};`,
+              );
+            }
+            break;
+          case ExportClosureMapping.NAMED_CONSTANT:
+            exportedConstants.push(key);
+            code = code.replace(`window.${key}=`, `const ${key}=`);
+            break;
+          default:
+            this.context.warn(
+              'Rollup Plugin Closure Compiler could not restore all exports statements.',
+            );
+            break;
+        }
+      });
+
+      if (exportedConstants.length > 0) {
+        if (code.endsWith('\n')) {
+          code = code.substr(0, code.lastIndexOf('\n'));
+        }
+        code += `export {${exportedConstants.join(',')}};`;
+      }
+    }
+
+    // TODO(KB): Sourcemaps fail :(
```

Comment:
magic-string might be useful in retaining maps

Reply:
@Rich-Harris strikes again. 

Good idea. https://github.com/Rich-Harris/magic-string